GoogleForCreators/web-stories-wp
# Design System: Popup Updates to Eliminate Redundant Custom Popup

## Context

There were a few places in the editor where a different `Popup` (internal to the editor) was getting used. This updates the shared `Popup` in the design system to consider these needs so we can eliminate the extra implementation. 

## Summary

Pretty much the editor's implementation of Popup existed because it needs to know if the editor's in RTL mode and find the topOffset of its CMS. 

## Relevant Technical Choices

New props for the design system's Popup are: `isRTL, topOffset, invisible` 
- invisible is only ever used in the color input.
- topOffset (which is a bummer of a name because we have an internal `$offset` that's getting used to decide positioning but I didn't change that) is for when a popup is maybe going to reach the top of the page so it needs an extra barrier because the found y min is actually wrong (app starts, say, 32px below the top of the browser window. 
- isRTL flips the y transforms as needed 

The style updates to the Popup's container mean that the scrollbar css isn't needed anymore.

## To-do



## User-facing changes

| Component | Staging - LTR | Staging - RTL | New - LTR | New - RTL |
| --- | --- | --- | --- | --- | 
| Page Attachment |  <img width="192" alt="link LTR" src="https://user-images.githubusercontent.com/10720454/152402232-8a8be55b-8240-43ff-a204-2217107857ca.png"> | <img width="189" alt="link RTL" src="https://user-images.githubusercontent.com/10720454/152402253-4fd40457-30d5-44b7-8b8f-669169fc45bb.png"> | <img width="307" alt="new link LTR" src="https://user-images.githubusercontent.com/10720454/152402308-67f6bd47-60ff-4b2c-afe1-05f29d7fc8d6.png"> | <img width="229" alt="new link RTL" src="https://user-images.githubusercontent.com/10720454/152402341-ec085123-02fb-4ad3-8aab-3a0f0019bd2f.png"> |
| Color Input | <img width="442" alt="color picker LTR" src="https://user-images.githubusercontent.com/10720454/152402487-c41f39a9-0f8c-45e4-b596-9555b6546ec8.png"> | <img width="335" alt="color picker RTL" src="https://user-images.githubusercontent.com/10720454/152402491-d3035627-0742-4e7c-86e9-f32c32919a49.png"> | <img width="413" alt="new color picker LTR" src="https://user-images.githubusercontent.com/10720454/152402493-824a6550-3abf-4b6c-8f8f-e43d03b56edf.png"> | <img width="351" alt="new color picker RTL" src="https://user-images.githubusercontent.com/10720454/152402494-bba1f9a2-4829-4b0d-9383-b4b7691f6b11.png"> |
| Play Pause | <img width="230" alt="play LTR" src="https://user-images.githubusercontent.com/10720454/152402575-ae59313f-9bc5-4d92-8cf3-b43e9c084802.png"> | <img width="234" alt="play RTL" src="https://user-images.githubusercontent.com/10720454/152402577-79310073-57a3-4f6d-b41c-c4b8bfe5534d.png"> | <img width="226" alt="new play LTR" src="https://user-images.githubusercontent.com/10720454/152402578-a735b7f8-840a-41e1-8ff5-f7e1b2272d9f.png"> | <img width="257" alt="new play RTL" src="https://user-images.githubusercontent.com/10720454/152402579-e79cbb6b-783b-4a87-89dc-a97201ab35b2.png"> | 
| Style Presets | <img width="632" alt="more styles LTR" src="https://user-images.githubusercontent.com/10720454/152402663-c3590a20-0228-4a3d-8051-8d53df328d66.png"> | <img width="512" alt="more styles RTL" src="https://user-images.githubusercontent.com/10720454/152402667-d14817f7-7734-496b-9f24-7c47f6e9ab74.png"> | <img width="536" alt="new more styles LTR" src="https://user-images.githubusercontent.com/10720454/152402703-fa1a79d5-cd7e-4a1b-9790-5f0ec8de05cf.png"> | <img width="365" alt="new more styles RTL" src="https://user-images.githubusercontent.com/10720454/152402706-75c8cfa9-e052-4b1f-bd0f-6fcb97720290.png"> | 

So, hopefully, if i did it right - no changes. 

## Testing Instructions

Unfortunately, the `Popup` in the design system touches a bunch of stuff. Regression testing on the following: 
- Tooltip behavior (be mindful of tooltips that change location when a button is interacted with while they are open - like locked vs unlocked), scroll, when expected space isn't available) 
- Drop downs 
- Editor: style presets, color input, play pause button, links in canvas
- Dashboard: stories view tooltip toggle 




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10089
